articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,313 @@
+
+Since the forward slash (`/`) is replaced with double underscores (`__`) when translating to DDS from ROS topic names, two characters out of the 256 character limit are lost with each additional namespace.
+One proposed alternative was to add a constraint that ROS topic names could not use capital letters, and then capital letters could be used as the stand in for the forward slashes (`/`).
+
+Trade-offs:
+
+- Uses one less character per namespace and makes it easier to calculate the maximum length of a ROS topic or service name.
+- Prevents users from using capital letters in their names, which is constraining and might be a problem for backwards compatibility with ROS 1 topics and services.
+
+Rational:
+
+Preventing users from using capital letters was too constraining for the added benefit.
+
+#### Prefix with Double Underscores
+
+This alternative differs only in that it uses two underscores in the prefix, i.e. `rt__` rather than `rt_`.
+
+Trade-offs:
+
+- More consistent with replacement of other forward slashes (`/`)
+- Uses one more character
+
+Rational:
+
+Slight preference given to the shorter alternative.
+
+#### Limited Prefixes Alternative
+
+This alternative would:
+
+- not prefix topics
+- optionally prefix other kinds of "implementation detail topics"
+
+Trade-offs:
+
+- it would be easier to have ROS subscribe to a DDS created topic
+
+  - e.g. DDS publisher on topic `image` could be subscribed to in ROS using just `image`
+  - however, the types would need to match as well
+  - in the current proposal the ROS topic `image` would become `rt_image`, so DDS topics would need to follow the ROS topic conversion scheme to interoperate with ROS components
+
+- it would be hard to distinguish ROS created DDS topics and normal DDS topics
+
+- services would still need to be differentiated
+
+  - e.g. service `/foo` would need to make two topics, something like `foo_Request` and `foo_Reply`
+
+Rational:
+
+Slight preference was given to easily categorizing ROS created topics with DDS created topics over easily connecting to existing DDS topics.
+Connecting to DDS topics could be achieved by having an option when subscribing or publishing to "alias" to an implementation topic name, e.g. something like `sub->alias_to_explicit_topic('dds_topic')`.
+Also, the workaround for separating ROS created topics from other DDS topics was considered to be more complicated than suggested solution for allowing users to specify specific DDS topic names for their publishers and subscriptions.
+
+#### Suffix Alternative
+
+This alternative would:
+
+- No prefix
+
+- The prefix is restructured to be a suffix, i.e. `rX_<topic>` -> `<topic>_rX_`
+
+  - this would be unique to user defined names because they cannot have a trailing underscore (`_`)
+
+Trade-offs:
+
+- more difficult to distinguish ROS created DDS topics from normal or built-in DDS topics when listing them using DDS tools because they are not sorted by a prefix
```

Comment:
I don't think that's the opposite. This line is talking about distinguishing between DDS topics and ROS created DDS topics.

I think you're thinking about yet another alternative which is `r_<topic>_X_`, which keeps ROS stuff together, but sorts first by namespace and then by type. You can add another Alternative section if you like or push for that.

My opinion is that anything we choose will be preferring one sorting over another, e.g. namespace vs type vs ROS related or not, and no matter which we pick, someone at some point will want to do the other. So I place more importance on the other aspects of the alternatives like number of characters used, ability to prevent collisions by design, and obviousness to unfamiliar observers.